finality-aleph/src/network/mod.rs

```diff
@@ -89,8 +89,11 @@ pub trait PeerId: PartialEq + Eq + Clone + Debug + Display + Hash + Codec + Send
 pub trait AddressingInformation: Debug + Hash + Codec + Clone + Eq + Send + Sync + 'static {
     type PeerId: PeerId;
 
-    /// Returns the peer id associated with this multiaddress if it exists and is unique.
+    /// Returns the peer id associated with this address.
     fn peer_id(&self) -> Self::PeerId;
+
+    /// Checks whether the information is valid.
+    fn valid(&self) -> bool;
```

Comment:
nit: In implementation this is more of verification of signature than checking if address is valid. So maybe verify?

Reply:
Yup, makes more sense, fixed.